### ISSUE TITLE
escaping password for Postgress connection

### ISSUE THREAD

**Issue by jnoortheen** [django-dbbackup/django-dbbackup#301]
I understand the reason behind this escaping, but this creates problem for passwords with special characters like `get_escaped_command_arg("pass!word")` -> `"'pass!word'"`

That extra pair of quotes make the command fail. Is it possible to remove the line of code safely?

Location of the code: `dbbackup/db/postgresql.py:19`

**Comment by JamesHannon** [comment]
Just ran into this issue myself ... specifically Windows 10. I don't experience it at all in my Unix/Linux system.

I solved it by using

`def get_escaped_command_arg(arg):
    return arg if os.name == 'nt' else quote(arg)`

Line 427/428 in dbbackup.utils.py

**Comment by adi-** [comment]
Doesn't work. What the `quote` function returns is i.e. `{'PGPASSWORD': "'my_password'"}` which fails to authenticate

**Comment by adi-** [comment]
What I suspect is env variable is already escaped with double quotes. I guess there shouldn't be `get_escaped_command_arg(arg)` used here https://github.com/django-dbbackup/django-dbbackup/blob/master/dbbackup/db/postgresql.py#L19

**Comment by silviogutierrez** [comment]
I thought this was related to special characters, but it seems like this actually adds quotes to any password and thus all Postgres passwords fail? Could that be the case?

**Comment by silviogutierrez** [comment]
Upon more research, that's almost certainly the case: this code:

https://docs.python.org/3/library/shlex.html#shlex.quote

is meant to be used in command arguments, not as an environment variable. Because we're setting an environment variable, the quotes are interpreted literally.

I'm pretty sure this means Postgres authentication is entirely broken right now, which is surprising since this is a pretty popular library.

Of course, up until recently, I had no password in Postgres as it was a local db.

Hope that helps someone. Great library nonetheless!

**Comment by jonathan-s** [comment]
@silviogutierrez Do you have a chance to verify if that's the case?
> but it seems like this actually adds quotes to any password and thus all Postgres passwords fail?

**Comment by silviogutierrez** [comment]
@jonathan-s : yes, that's the case. I fixed it in my Docker builds by running this beauty:

```
# See: https://github.com/django-dbbackup/django-dbbackup/issues/301
RUN sed -i "s/utils\.get_escaped_command_arg(self\.settings\['PASSWORD'\])/self.settings['PASSWORD']/" /usr/local/lib/python3.8/site-packages/dbbackup/db/postgresql.py
```

**Comment by jonathan-s** [comment]
Thanks! @silviogutierrez I'll make sure that this gets fixed in the next release!

**Comment by jonathan-s** [comment]
@silviogutierrez Do you think you could try current master and see if you see this issue? I think this should be fixed.

**Comment by jonathan-s** [comment]
Should be fixed in #361, if you have any issues in current master let me know and we'll sort this out.

**Comment by silviogutierrez** [comment]
@jonathan-s : confirmed, it works. Thanks for your work on this library.
